jdaviz/core/tools.py

```diff
@@ -354,28 +354,88 @@ def activate(self):
         self.viewer.add_event_callback(self.on_mouse_event,
                                        events=['dragmove', 'click'])
         for mark in self.viewer.figure.marks:
-            if np.any([x in mark.labels for x in ('vmin', 'vmax')]):
+            if np.any([x in mark.labels for x in ('vmin', 'vmax', 'stretch_knots')]):
                 mark.colors = ["#c75d2c"]
 
     def deactivate(self):
         self.viewer.remove_event_callback(self.on_mouse_event)
         for mark in self.viewer.figure.marks:
-            if np.any([x in mark.labels for x in ('vmin', 'vmax')]):
+            if np.any([x in mark.labels for x in ('vmin', 'vmax', 'stretch_knots')]):
                 mark.colors = ["#007BA1"]
 
     def on_mouse_event(self, data):
         if (time.time() - self._time_last) <= 0.05:
             # throttle to 200ms
             return
 
+        # Mouse event coordinates
         event_x = data['domain']['x']
-        current_bounds = [self.viewer._plugin.stretch_vmin_value,
-                          self.viewer._plugin.stretch_vmax_value,]
-        att_names = ["stretch_vmin_value", "stretch_vmax_value"]
-        closest_bound_ind = np.argmin([abs(current_bounds[0] - event_x),
-                                       abs(current_bounds[1] - event_x)])
+        event_y = data['domain']['y']
+
+        x_min, x_max = self.viewer.state.x_min, self.viewer.state.x_max
+        y_min, y_max = self.viewer.state.y_min, self.viewer.state.y_max
 
-        setattr(self.viewer._plugin, att_names[closest_bound_ind], event_x)
+        current_bounds = [self.viewer._plugin.stretch_vmin_value,
+                          self.viewer._plugin.stretch_vmax_value]
+
+        distances_to_bounds = [abs(current_bounds[0] - event_x), abs(current_bounds[1] - event_x)]
+        closest_bound_index = np.argmin(distances_to_bounds)
+        closest_bound_distance = distances_to_bounds[closest_bound_index]
+
+        stretch_type = self.viewer._plugin.stretch_function_value
+        if stretch_type == 'spline':
+            # Retrieve current knots and bounds
+            # knot_x is in units of the x-axis (pixel value)
+            knot_x = self.viewer._plot.marks['stretch_knots'].x
+            # knot_y is in units of the (hidden) normalized y-axis from 0 to 1
+            knot_y = self.viewer._plot.marks['stretch_knots'].y
+
+            # event_y is in units of the y-axis (density),
+            # so we need to rescale event_y to the same units as knot_y
+            event_y_normalized = (event_y - y_min) / (y_max - y_min)
+            # Drop event if the event coordinate are out of bounds
+            if not 0 <= event_y_normalized <= 0.92:
```

Comment:
I'd expect this to be the same value used for normalizing the curve to force it below the colormap.  Otherwise values above 0.9 will map to a stretch value above 1.

```suggestion
            if not 0 <= event_y_normalized <= 0.9:
```

Reply:
> It was an attempt to position the knots a bit closer to the color bar.

Then the 0.92 should replace other 0.9s (maybe we should have a global variable for this value that we re-use in all instances now that it appears in a few places?)

> Do you know if there's a way to select the color bar location dynamically from the histogram viewer

Not really, it's placed at the top (y=1 on the normalized axis) but has a `stroke_width` that I don't think we can easily map back to the y-axis as it depends probably on the height of the viewer.  So maybe we need comments where the stroke width and viewer heights are set, so if we change them in the future we'll revisit the scaling value.